### PR TITLE
Nodejs Changes

### PR DESCRIPTION
**Pull request by ravivangara17**

### CODE REVIEW

File: components/cookbooks/node/metadata.rb [hunk 1]
```diff
@@ -3,7 +3,7 @@
 version          "0.1"
 maintainer       "OneOps"
 maintainer_email "support@oneops.com"
-license          "Apache License, Version 2.0"
+license          "Copyright OneOps, All rights reserved."
```
Comment (schwankl) [inline components/cookbooks/node/metadata.rb]: its apache licensed... why did you change? revert

File: components/cookbooks/node/metadata.rb [hunk 2]
```diff
@@ -17,12 +17,13 @@
   :packages => [ 'base', 'mgmt.catalog', 'catalog', 'mgmt.manifest', 'manifest', 'bom' ]
 
 attribute 'install_method',
-  :description => "Install via (source or package or binary)",
+  :description => "Install via (package or binary)",
```
Comment (schwankl) [inline components/cookbooks/node/metadata.rb]: there should be a source option - there is a source recipe

File: components/cookbooks/node/metadata.rb [hunk 3]
```diff
@@ -31,16 +32,17 @@
   :format => {
     :category => '1.Global',
     :order => 2,
-    :help => 'Version of Node.js' , 
-    :form => { 'field' => 'select', 'options_for_select' => [['0.10.33','0.10.33'],['0.12.4','0.12.4']] }
+    :help => 'Version of Node.js' ,
+    :form => {'field' => 'select', 'options_for_select' => [['0.10.33','0.10.33'],['0.10.41','0.10.41'],['0.12.4','0.12.4'],['4.2.4','4.2.4']]}
   }
 
 attribute 'src_url',
```
Comment (schwankl) [inline components/cookbooks/node/metadata.rb]: src url is for building from source

File: components/cookbooks/node/recipes/install_from_source.rb [hunk 4]
```diff
@@ -18,7 +18,7 @@
 # limitations under the License.
 #
 
-include_recipe "build-essential"
+#include_recipe "build-essential"
```
Comment (schwankl) [inline components/cookbooks/node/recipes/install_from_source.rb]: why commented out?

File: packs/node.rb [hunk 5]
```diff
@@ -0,0 +1,225 @@
...
+               'up' => metric(:unit => '', :description => 'Status', :dstype => 'GAUGE')
+             },
+             :thresholds => {
+
+             }
+           }
+         }
+
+
+configure_nodeModule_command=  <<-"EOF"
```
Comment (schwankl) [inline packs/node.rb]: this should not be in the pack

File: packs/node.rb [hunk 6]
```diff
@@ -0,0 +1,225 @@
...
+             :thresholds => {
+               'LowDiskSpace' => threshold('1m','avg','space_used',trigger('>=', 90, 5, 2), reset('<', 85, 5, 1)),
+               'LowDiskInode' => threshold('1m','avg','inode_used',trigger('>=', 90, 5, 2), reset('<', 85, 5, 1))
+             },
+           }
+         }
+
+# depends_on
+[
+  {:from => 'user-app', :to => 'compute'},
```
Comment (schwankl) [inline packs/node.rb]: should depend on os

File: components/cookbooks/node_module/recipes/add.rb [hunk 7]
```diff
@@ -13,7 +13,7 @@
 node.set['node_module']['npm'] = `which npm`.strip
 node.set['node_module']['node'] = `which node`.strip
 
-execute "#{node['node_module']['npm']} install #{module_name}@#{module_version} --production" do
+execute "#{node['node_module']['npm']} install #{module_name}@#{module_version}" do
```
Comment (schwankl) [inline components/cookbooks/node_module/recipes/add.rb]: why remove --production ? sounds like its useful:
https://futurestud.io/blog/how-to-install-npm-dependencies-for-production-environment

File: components/cookbooks/node/metadata.rb (hunk 2 above)
Comment (ravivangara17) [inline components/cookbooks/node/metadata.rb]: we are going to remove the source recipe, we will be only using package and binary.

File: components/cookbooks/node/metadata.rb (hunk 1 above)
Comment (ravivangara17) [inline components/cookbooks/node/metadata.rb]: This change was made in the internal Walmart labs repo, would you like us to update it with Apache license?

File: components/cookbooks/node/metadata.rb (hunk 2 above)
Comment (ravivangara17) [inline components/cookbooks/node/metadata.rb]: We are supporting only package and binary for now.

File: components/cookbooks/node/metadata.rb (hunk 3 above)
Comment (ravivangara17) [inline components/cookbooks/node/metadata.rb]: location of node JS binary.

File: components/cookbooks/node/recipes/install_from_source.rb (hunk 4 above)
Comment (ravivangara17) [inline components/cookbooks/node/recipes/install_from_source.rb]: removing this file, we do not support install from source at this time.

File: packs/node.rb (hunk 5 above)
Comment (ravivangara17) [inline packs/node.rb]: removed the CONFIGURE lines from PACK , BTW these lines are already there in some of our existing packs in our current circuits.

File: components/cookbooks/node/metadata.rb (hunk 1 above)
Comment (schwankl) [inline components/cookbooks/node/metadata.rb]: yes, it was already released w/ apache license

File: components/cookbooks/node_module/recipes/add.rb (hunk 7 above)
Comment (ravivangara17) [inline components/cookbooks/node_module/recipes/add.rb]: Production is internal one, we leave it generic

Comment (schwankl) [inline components/cookbooks/node_module/recipes/add.rb]: that doesnt make sense
